csm_web/scheduler/email/email_utils.py

```diff
@@ -0,0 +1,242 @@
+import os
+import base64
+import html
+from email.message import EmailMessage
+
+from django.conf import settings
+
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+from .email_errors import EmailFormattingError, NoEmailError, EmailAuthError, email_error_handling
+
+PROFILE_URL = "https://scheduler.csmentors.org/profile"
+
+# Email error handling wrapper adds logger argument as last argument
+
+
+@email_error_handling
+def email_enroll(student):
+    """Sends the corresponding mentor an email notification that the STUDENT enrolled in their section"""
+
+    try:
+        mentor = student.section.mentor
+        course_title = student.course.title
+        mentor_name = mentor.user.first_name + " " + mentor.user.last_name
+        student_name = student.user.first_name + " " + student.user.last_name
+        mentor_email = mentor.user.email
+        student_email = student.user.email
+    except AttributeError:
+        raise NoEmailError
+
+    if mentor.user.subscribed:
+        body = _render_template("student_enroll/notif.html").format(
+            course_title=html.escape(course_title),
+            mentor_name=html.escape(mentor_name),
+            student_name=html.escape(student_name),
+        )
+        _email_send_message(f"[CSM - {course_title}] Student Enrolled", body, mentor_email)
+
+    if student.user.subscribed:
+        body = _render_template("student_enroll/confirm.html").format(
+            course_title=html.escape(course_title),
+            mentor_name=html.escape(mentor_name),
+            student_name=html.escape(student_name),
+        )
+        _email_send_message(f"[CSM - {course_title}] Section Enrollment Confirmation", body, student_email)
```

Comment:
Looking at these again, is there any possibility of XSS or other security vulnerabilities from this? i.e. is `html.escape` sufficient?

This OWASP page may be useful to read through: https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html

Though this may be more special since it's in an email, and from [this stackoverflow question](https://stackoverflow.com/questions/3054315/is-javascript-supported-in-an-email-message), it does seem like most modern email clients don't support JS at all (and would remove it from the email content), though there is no general specification about it.

It's an interesting thing to think about though, and perhaps will become more relevant later when we allow users to customize their names, since then the mentor and student names will be fully user controlled (whereas now they're automatically taken from their Google accounts).